src/nvim/path.c

```diff
@@ -2186,9 +2186,16 @@ static int path_get_absolute_path(const char_u *fname, char_u *buf,
 
 /// Check if the given file is absolute.
 ///
-/// This just checks if the file name starts with '/' or '~'.
 /// @return `TRUE` if "fname" is absolute.
 int path_is_absolute_path(const char_u *fname)
 {
+#ifdef WIN32
+  // A name like "d:/foo" and "//server/share" is absolute
+  return (isalpha(fname[0]) && fname[1] == ':'
```

Comment:
I would prefer to deal with this one separately. The use of isalpha is similar to what is done in vim/vim and there are a least a couple other locations, unrelated to this PR, where isalpha is used.